internal/services/synapse/synapse_pipeline_resource.go

```diff
@@ -0,0 +1,361 @@
+
+	workspaceId, err := parse.WorkspaceID(d.Get("synapse_workspace_id").(string))
+	if err != nil {
+		return err
+	}
+
+	client, err := synapseClient.PipelinesClient(workspaceId.Name, environment.SynapseEndpointSuffix)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] preparing arguments for Synapse Pipeline creation.")
+
+	id := parse.NewPipelineID(workspaceId.SubscriptionId, workspaceId.ResourceGroup, workspaceId.Name, d.Get("name").(string))
+	if d.IsNewResource() {
+		existing, err := client.GetPipeline(ctx, id.Name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for present of existing %s: %+v", id, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_synapse_pipeline", id.ID())
+		}
+	}
+
+	pipeline := &artifacts.Pipeline{
+		Parameters:  expandSynapseParameters(d.Get("parameters").(map[string]interface{})),
+		Variables:   expandSynapseVariables(d.Get("variables").(map[string]interface{})),
+		Description: utils.String(d.Get("description").(string)),
+	}
+
+	if v, ok := d.GetOk("activities_json"); ok {
+		activities, err := deserializeSynapsePipelineActivities(v.(string))
+		if err != nil {
+			return fmt.Errorf("parsing 'activities_json' for %s", err)
+		}
+		pipeline.Activities = activities
+	}
+
+	if v, ok := d.GetOk("annotations"); ok {
+		annotations := v.([]interface{})
+		pipeline.Annotations = &annotations
+	} else {
+		annotations := make([]interface{}, 0)
+		pipeline.Annotations = &annotations
+	}
+
+	config := artifacts.PipelineResource{
+		Pipeline: pipeline,
+	}
+
+	future, err := client.CreateOrUpdatePipeline(ctx, id.Name, config, "")
+	if err != nil {
+		return fmt.Errorf("creating %s: %+v", id, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting on creation/updation for %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceSynapsePipelineRead(d, meta)
+}
+
+func resourceSynapsePipelineRead(d *schema.ResourceData, meta interface{}) error {
+	synapseClient := meta.(*clients.Client).Synapse
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+	environment := meta.(*clients.Client).Account.Environment
+
+	id, err := parse.PipelineID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	client, err := synapseClient.PipelinesClient(id.WorkspaceName, environment.SynapseEndpointSuffix)
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.GetPipeline(ctx, id.Name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			log.Printf("[DEBUG] Synapse Pipeline %q was not found - removing from state!", d.Id())
+			return nil
+		}
+		return fmt.Errorf("reading %s: %+v", id, err)
+	}
+
+	workspaceId := parse.NewWorkspaceID(id.SubscriptionId, id.ResourceGroup, id.WorkspaceName).ID()
+	d.Set("synapse_workspace_id", workspaceId)
```

Comment:
We can combine them together for simplicity:

```suggestion
	d.Set("synapse_workspace_id", parse.NewWorkspaceID(id.SubscriptionId, id.ResourceGroup, id.WorkspaceName).ID())
```

Reply:
Fixed